### PR TITLE
Add a concrete governance model

### LINKED ISSUE #1064: Add governance and membership pathway documentation
**Is your feature request related to a problem? Please describe.**
We want to make it clear how community members can become maintainers of the project. We also want to incorporate aspects of the argoproj [governance](https://github.com/argoproj/argoproj/blob/fd0748186cac552b4a89228a541c9c855e4434cf/community/GOVERNANCE.md) and [membership](https://github.com/argoproj/argoproj/blob/fd0748186cac552b4a89228a541c9c855e4434cf/community/membership.md) docs

**Describe the solution you'd like**
We'd like to reflect Argo's membership [roles](https://github.com/argoproj/argoproj/blob/main/community/membership.md) - we would only need a subset to begin with, including:
* Member - allowing triaging of issues and can be assigned issues/PRs
* Approver - allowing merges to main
* Maintainer (aka "Lead") - setting direction and priorities. This includes voting powers related to HEPs (as described [here](https://github.com/argoproj-labs/hera/blob/345e687315920f59cc7a4600a975e3cd026dedaa/proposals/README.md))

We should require a certain number of contributions to become a Member and Approver. Becoming a Maintainer should be with nomination and agreement between existing maintainers.

**Describe alternatives you've considered**
* Not expanding the pool of maintainers
* Ad-hoc giving people permissions

**Additional context**

### PR DESCRIPTION
**Pull request by samj1912**
**Pull Request Checklist**
- [x] Fixes #1064 

**Description of PR**

This adds a governance process to hera and a pathway to maintainership.

### CODE REVIEW

**Comment by samj1912** [comment]
@flaviuvadan / @elliotgunton this will require approvals from both of you to merge.

File: MEMBERS.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
+# Hera Project Members
+
+## TOC
+
 - Flaviu Vadan <flaviu.vadan@dynotx.com>
```
Comment (samj1912) [inline MEMBERS.md:5]: @flaviuvadan pls provide your personal email address here.

File: GOVERNANCE.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+- Be a committer to Hera for at-least 2 months
+- Reviewer for or author of at least 10 substantial PRs to the codebase, with the definition of substantial subject to the TOC's discretion (e.g. refactors, enhancements rather than grammar correction or one-line pulls).
+- Exhibiting sound technical judgment through PR contributions
+- Exhibiting sound technical judgment through PR reviews
+- Be nominated by an existing maintainer/TOC member and be voted in by remaining maintainers/TOC members (via a lazy-consensus vote lasting 2 weeks)
+
+#### Privileges
+
+Maintainers will have the following permissions:
+- Maintain permissions on the Hera Github repository
```
[the remark this answers is not in the document]
Comment (samj1912) [reply GOVERNANCE.md:68]: Nope - that is limited to Admins

[the remark this answers is not in the document]
Comment (samj1912) [reply GOVERNANCE.md:68]: Noted at https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization#permissions-for-each-role

File: GOVERNANCE.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
+# Hera Governance
+
+This document outlines the governance for the overall Hera Project.
+
+Any changes to Hera's governance and this document require a unanimous vote by all TOC members.
+
+All votes mentioned in this document follow a [lazy-consensus process](https://medlabboulder.gitlab.io/democraticmediums/mediums/lazy_consensus/) with a default time-period of 2 weeks unless they are mentioned to be unanimous in which case they require unanimous consensus from all binding voters.
+
+## Roles and Membership
```
[the remark this answers is not in the document]
Comment (samj1912) [reply GOVERNANCE.md:9]: Would prefer to do that in a separate PR

File: GOVERNANCE.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+- Be an approver to Hera for at least 2 months
+- Reviewer for or author of at least 10 substantial PRs to the codebase, with the definition of substantial subject to the TOC's discretion (e.g. refactors, enhancements rather than grammar correction or one-line pulls).
+- Exhibiting sound technical judgment through PR contributions
+- Exhibiting sound technical judgment through PR reviews
+- Be nominated by an existing maintainer/TOC member and be voted in by remaining maintainers/TOC members (via a lazy-consensus vote lasting 2 weeks)
+
+#### Privileges
+
+Maintainers will have the following permissions:
+- Ability to update permissions on the Hera Github repository
```
Comment (elliotgunton) [inline GOVERNANCE.md:68]:
```suggestion
- ["Maintain" permissions](https://docs.github.com/en/organizations/managing-user-access-to-your-organizations-repositories/managing-repository-roles/repository-roles-for-an-organization#permissions-for-each-role) on the Hera Github repository
```

File: MEMBERS.md (hunk 1 above)
Comment (flaviuvadan) [reply MEMBERS.md:5]:
```suggestion
- Flaviu Vadan <flaviuvadan@gmail.com>
```

File: GOVERNANCE.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+- Defining the governance structure of the project
+- Appointing maintainers and leads
+- Defining the [HEP](proposals/README.md) process through which cross-cutting changes are proposed and approved
+- Defining the annual roadmap
+- Responsible for handling security reports and incidents
+- Anything else that falls through the cracks
+
+#### Nomination Process
+
+For a new TOC member to be appointed they must -
```
Comment (flaviuvadan) [inline GOVERNANCE.md:28]:
```suggestion
For a new TOC member to be appointed they must:
```

File: GOVERNANCE.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+- Ensuring contributions align with project goals and meet the project's quality standards
+- Reviewing, approving, and merging PRs
+- Planning release milestones, and releasing components under the team's area of responsibility
+- Representing the work of the team to the community
+- Supporting contributors
+- Growing the team by mentoring aspiring contributors and maintainers
+
+#### Nomination Process
+
+For a new maintainer to be appointed they must -
```
Comment (flaviuvadan) [inline GOVERNANCE.md:57]:
```suggestion
For a new maintainer to be appointed they must:
```

**Review by flaviuvadan — APPROVED** [review]
I agree with the model and do not have anything to add at the moment! Thank you for writing this up :)

**Review by elliotgunton — APPROVED** [review]
LGTM! 🚀

**Pull request merged**